interface/views.py

```diff
@@ -92,9 +93,13 @@ def upload(request, course_pk, assignment_pk):
             else:
                 return redirect(users_list, course_pk, assignment_pk)
 
-            return render(request, 'interface/upload.html', {
-                'form': form,
-            })
+        elif request.FILES.get('file_size_warning', None):
```

Comment:
Q: Isn't something like this cleaner? (you might have investigated this)
https://stackoverflow.com/questions/38842053/how-to-check-the-file-size-limit-in-django

Reply:
Thinking that we can inherit the UploadFileForm and add that function

Reply:
The process of uploading a file is something similar to:

1. Client selects the file
2. The upload handlers save the file on our system
3. The form data validation

The resource you pointed to, it's cleaner, but the check happens on the third stage where we already have taken the time and resources to get the entire file on our server. Maybe I can clear this mess by returning a fake `UploadedFile` with no data, but with the size of infinity so that we can use the restriction you pointed to. I will look into it!